src/sentry/sentry_metrics/multiprocess.py

```diff
@@ -350,54 +352,56 @@ def process_messages(
     metrics = get_metrics()
 
     strings = set()
-    parsed_payloads_by_offset = {
-        msg.offset: json.loads(msg.payload.value.decode("utf-8"), use_rapid_json=True)
-        for msg in outer_message.payload
-    }
-
-    for message in parsed_payloads_by_offset.values():
-        metric_name = message["name"]
-        tags = message.get("tags", {})
-
-        parsed_strings = {
-            metric_name,
-            *tags.keys(),
-            *tags.values(),
+    with metrics.timer("process_messages.parse_outer_message"):
+        parsed_payloads_by_offset = {
+            msg.offset: json.loads(msg.payload.value.decode("utf-8"), use_rapid_json=True)
+            for msg in outer_message.payload
         }
-        strings.update(parsed_strings)
+
+        for message in parsed_payloads_by_offset.values():
+            metric_name = message["name"]
+            tags = message.get("tags", {})
+
+            parsed_strings = {
+                metric_name,
+                *tags.keys(),
+                *tags.values(),
+            }
+            strings.update(parsed_strings)
 
     with metrics.timer("metrics_consumer.bulk_record"):
```

Comment:
Let's have a metric with the number of strings we are passing to the indexer. That will tell us how effective batching is at reducing the amount of strings we need to resolve (as we exclude the duplicates).